pkg/forward/forward.go

```diff
@@ -416,6 +432,87 @@ func (isdf *InterStepDataForward) forwardAChunk(ctx context.Context) {
 	forwardAChunkProcessingTime.With(map[string]string{metrics.LabelVertex: isdf.vertexName, metrics.LabelPipeline: isdf.pipelineName, "buffer": isdf.fromBuffer.GetName()}).Observe(float64(time.Since(start).Microseconds()))
 }
 
+// streamMessage streams the data messages to the next step.
+func (isdf *InterStepDataForward) streamMessage(
+	ctx context.Context,
+	dataMessages []*isb.ReadMessage,
+	processorWM wmb.Watermark,
+) (map[string][]isb.Offset, error) {
+	// create space for writeMessages specific to each step as we could forward to all the steps too.
+	var messageToStep = make(map[string][]isb.Message)
+	for buffer := range isdf.toBuffers {
+		// over allocating to have a predictable pattern
+		messageToStep[buffer] = make([]isb.Message, 0, len(dataMessages))
+	}
+
+	if isdf.opts.vertexType == dfv1.VertexTypeSource {
+		errMsg := "stream is not supported for source data transformer"
+		isdf.opts.logger.Errorw(errMsg)
+		return nil, fmt.Errorf(errMsg)
+	}
+
+	if len(dataMessages) > 1 {
+		errMsg := "data message size is not 1 with UDF streaming"
+		isdf.opts.logger.Errorw(errMsg)
+		return nil, fmt.Errorf(errMsg)
+	} else if len(dataMessages) == 1 {
```

Comment:
nit: this else if not needed.

Reply:
just extra check in case some bug is introduced to allow batch size larger than 1 in future.